GetJobber/atlantis
# feat: Added Pre Release Action

## Motivations

 When updating Atlantis one of the major barriers to contributing is seeing how the changes you are making interact with other projects such as Jobber Online. The reason this can be scary is `npm link` doesn't work with Atlantis and the workaround for getting the changes to work isn't well known and can be confusing.

To resolve this we should allow contributors to generate Pre-releases to make testing changes easier for both contributors and reviewers

## Changes



### Added

- feat: Added ability to generate pre-releases through Github actions
- feat: Added Github action to comment on the PR for the branch to make getting build information easier

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

Testing Pre-release general
 - Branch off of this branch with the name `<your-name>-pre-release-review`
 - Create a draft PR for that branch (to verify comments work)
 - Create changes in `packages/components` and `packages/design` See: e41b90217e2718a0f322da43b024d98c771c36ad
 - Go to actions and generate a Pre-release with the `Lerna Publish` setting for the branch
 - Observe that a pre-release is generated and commented on the draft PR

Test Pre-release unhappy path:
 - After the pre-release is publish from the previous step, run another pre-release
 - Observe that when the workflow finishes a comment will be updated to link to the logs and show the previous build's information

Test pre-release force:
 - add dependencies to the package.json of `components` and `design`
   - You can run `npm install is-even -w @jobber/components` and `npm install is-even -w @jobber/design` to do this
 - Commit and push
 - Run a pre-release with the setting `Lerna Force Publish Components & Design`

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
